daemon/exec.go

```diff
@@ -183,8 +183,10 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, optio
 			ec.Lock()
 			ec.Container.ExecCommands.Delete(ec.ID)
 			ec.Running = false
-			exitCode := 126
-			ec.ExitCode = &exitCode
+			if ec.ExitCode == nil {
+				exitCode := 126
```

Comment:
We can probably use the consts here to give  more context what this magic number is; not sure if there's other logic in that file as well that could be useful here (just stepped away from my computer 🥹); https://github.com/moby/moby/blob/62120a727c7e28609fec296f23c59dda62078b8a/daemon/errors.go#L122-L125

Reply:
I'd like to refactor a lot of this code, but was still trying to look at whether it made sense to do that in this PR or a follow-up. I also wanted to see if there was another small change I could do here to address this issue in particular without possibly breaking a lot of other logic.

Reply:
The more I look at this, the more I dislike that `defer` error handling block. We capture a bunch of errors and default to `126` (`EACCES`) for any reason that we fail to start the exec. It's hard to have confidence changing anything here since so many different things could cause the `126` to be set, including failing to unmarshal a json spec, etc.